samples/extensions/vertex_dynamic_state/vertex_dynamic_state.cpp

```diff
@@ -350,19 +350,37 @@ void VertexDynamicState::build_command_buffers()
 		vkCmdBindDescriptorSets(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &descriptor_set, 0, nullptr);
 
 		/* skybox */
-		change_vertex_input_data(VERTEX_DYNAMIC_STATE_USE_FRAMEWORK_VERTEX_STRUCT);
-		vkCmdSetVertexInputEXT(draw_cmd_buffer, 1, &vertex_bindings_description_ext, 2, vertex_attribute_description_ext);
+		/* First set of vertex input dynamic data (Vertex structure) */
+		vertex_bindings_description_ext[0].stride  = sizeof(Vertex);
+		vertex_attribute_description_ext[1].offset = sizeof(float) * 3;
+		vkCmdSetVertexInputEXT(draw_cmd_buffer,
+		                       static_cast<uint32_t>(vertex_bindings_description_ext.size()),
+		                       vertex_bindings_description_ext.data(),
+		                       static_cast<uint32_t>(vertex_attribute_description_ext.size()),
+		                       vertex_attribute_description_ext.data());
+
 		vkCmdBindPipeline(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, skybox_pipeline);
 		draw_model(skybox, draw_cmd_buffer);
 
 		/* object */
-		vkCmdSetVertexInputEXT(draw_cmd_buffer, 1, &vertex_bindings_description_ext, 2, vertex_attribute_description_ext);
+		vkCmdSetVertexInputEXT(draw_cmd_buffer,
```

Comment:
Is this call to `vkCmdSetVertexInputEXT` needed?

Reply:
Good point. I checked it and `vkCmdSetVertexInputEXT` needs to be called when we want to change vertex input data only. Not before each `vkCmdBindPipeline`. Thanks for spotting this. I will remove unnecessary `vkCmdSetVertexInputEXT`.